Iterator/AMQPMessageIterator.php

```diff
@@ -65,7 +71,23 @@ public function current()
      */
     public function next()
     {
-        $this->wait();
+        if ($amqpMessage = $this->consumer->receive($this->timeout)) {
+            $this->AMQMessage = $amqpMessage;
+
+            $data = json_decode($this->AMQMessage->getBody(), true);
+            $data['body']['AMQMessage'] = $amqpMessage;
+
+            $message = new Message();
+            $message->setBody($data['body']);
+            $message->setType($data['type']);
+            $message->setState($data['state']);
+            $this->message = $message;
+
+            ++$this->counter;
+            $this->isValid = true;
+        } else {
+            $this->isValid = false;
```

Comment:
Please use a `return` statement and remove the `else`

Reply:
not sure I got it. I dont return anything from `next` method. The value is stored and used in another method: `valid`.

Reply:
I mean 
```php
    $this->isValid = true;
    return;
}

$this->isValid = false;